web/src/lib/components/faces-page/set-birth-date-modal.svelte

```diff
@@ -6,14 +6,32 @@
   import Icon from '$lib/components/elements/icon.svelte';
 
   export let birthDate: string;
+  let error = '';
 
   const dispatch = createEventDispatcher<{
     close: void;
     updated: string;
   }>();
 
+  // Calculate today's date in the local time zone and format it
+  const todayFormatted = new Date().toISOString().split('T')[0];
+
+  $: {
+    if (birthDate && birthDate > todayFormatted) {
+      // If the birthDate is set in the future, force the date to today's date
+      birthDate = todayFormatted;
+      error = 'Date of birth cannot be in the future.';
+    } else {
+      error = '';
+    }
+  }
+
   const handleCancel = () => dispatch('close');
-  const handleSubmit = () => dispatch('updated', birthDate);
+  const handleSubmit = () => {
+    if (!error) {
```

Comment:
The error message, was just to show to the user what went wrong. 
But I can update that to disable the button.

Reply:
@martabal I have updated the code, I suggest that we keep the error message, and disable the Set button in case of error. 
Also I have added the case where the user want to clear the birthdate. With my pervious changes, it will not be anymore possible once the birthdate is set.